basyx.submodelservice/basyx.submodelservice-feature-operation-dispatching/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/operationdispatching/execution/OperationExecutorProvider.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (C) 2024 DFKI GmbH (https://www.dfki.de/en/web)
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ * 
+ ******************************************************************************/
+package org.eclipse.digitaltwin.basyx.submodelservice.feature.operationdispatching.execution;
+
+/**
+ * @author Gerhard Sonnenberg DFKI GmbH
+ */
+@FunctionalInterface
+public interface OperationExecutorProvider {
```

Comment:
One idea would be to add the InvokableOperations to the Submodel when loading it via the GenericSubmodelFactory, instead of searching for the corresponding idShortPath in the OperationDispatcherSubmodelService.

This way no new feature is needed and we rely on the same mechanism already implemented in the Impl of AasService:

https://github.com/eclipse-basyx/basyx-java-server-sdk/blob/935edab02db757f13988ae37961ac629bb79ec77/basyx.submodelservice/basyx.submodelservice-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/InMemorySubmodelService.java#L236-L244

Let me know if this idea helps.